RetroBar/Utilities/ClockFlyoutLauncher.cs

```diff
@@ -0,0 +1,147 @@
+        {
+            Screen taskbarScreen = Screen.FromHandle(taskbarHwnd);
+            if (!GetWindowRect(taskbarHwnd, out Rect lpRect))
+            {
+                return;
+            }
+
+            _aeroClockInstance ??= Activator.CreateInstance(AeroClockType.Value);
+
+            if (ManagedShell.Common.Helpers.EnvironmentHelper.IsWindows8OrBetter)
+            {
+                ((IAeroClock)_aeroClockInstance).ShowFlyout(taskbarHwnd, ref lpRect);
+            }
+            else
+            {
+                ((IAeroClockLegacy)_aeroClockInstance).ShowFlyout(0, ref lpRect);
+            }
+
+            FixAeroClockFlyoutPosition(taskbarScreen);
+        }
+
+        private static void FixAeroClockFlyoutPosition(Screen taskbarScreen)
+        {
+            IntPtr clockFlyoutHwnd = FindWindow("ClockFlyoutWindow", null);
+            if (clockFlyoutHwnd == IntPtr.Zero)
+            {
+                return;
+            }
+
+            GetWindowRect(clockFlyoutHwnd, out Rect clockFlyoutRect);
+
+            // Keep the flyout at least 3 by 3 pixels inside the working area
+            int newX = (int)Math.Max(taskbarScreen.WorkingArea.Left + 3, Math.Min(taskbarScreen.WorkingArea.Right - (clockFlyoutRect.Width) - 3, clockFlyoutRect.Left));
+            int newY = (int)Math.Max(taskbarScreen.WorkingArea.Top + 3, Math.Min(taskbarScreen.WorkingArea.Bottom - (clockFlyoutRect.Height) - 3, clockFlyoutRect.Top));
+            SetWindowPos(clockFlyoutHwnd, IntPtr.Zero, newX, newY, 0, 0, (int)(NoPosFlags));
+        }
+
+        #endregion Aero clock
+
+        #region Modern clock
+
+        internal static void ShowClockFlyout()
+        {
+            // Current implementation is for the primary screen only
+            Screen screen = Screen.PrimaryScreen;
+            FlowDirection flowDirection = Application.Current.FindResource("flow_direction") as FlowDirection? ?? FlowDirection.LeftToRight;
+
+            PostMessage(WindowHelper.FindWindowsTray(WindowHelper.FindWindowsTray(IntPtr.Zero)), (uint)0x5CE, IntPtr.Zero, IntPtr.Zero);
+
+            // Launching the Modern clock is slower than the Aero clock, so we need to wait for the window to be created...
+            System.Windows.Threading.DispatcherTimer timer = new()
+            {
+                Interval = TimeSpan.FromTicks(10)
+            };
+
+            int attempts = 0;
+            StringBuilder className = new(256);
+
+            timer.Tick += (_, _) =>
+            {
+                if (++attempts > 100)
+                {
+                    timer.Stop();
+                    return;
+                }
+
+                IntPtr hwnd = GetForegroundWindow();
+                if (hwnd == IntPtr.Zero)
+                {
+                    return;
+                }
+
+                className.Clear();
+                _ = GetClassName(hwnd, className, className.Capacity);
+                if (className.ToString() != "Windows.UI.Core.CoreWindow")
```

Comment:
Is there any way we can more uniquely check this? There are a lot of modern windows with this class, and I was able to get this code to move them, for example by using the keyboard to activate another flyout at the same time.

Reply:
`GetForegroundWindow()` gets the foreground window that is brought up just after the clock is clicked with this option.

Reply:
Blame M$ for not using proper class names but generics XD

Reply:
Just found this Gist with C code that allows to "toggle the clock flyout in the Windows 10 taskbar on the monitor containing the mouse" (seems to be really complex, but it claims to be able to use other screens):
https://gist.github.com/valinet/bfc3e15d0d999cd3d2aab8d554955e6f

It seems that there's a program called "ExplorerPatcher", that allows to use the modern clock for the taskbar too: https://github.com/valinet/ExplorerPatcher

The Gist is also referenced here: https://github.com/valinet/ExplorerPatcher/discussions/67

Reply:
Overwriting memory in explorer.exe seems not great but that does look like a way to fix the multi-screen support. However, we would still need this loop to move the window to the proper edge, and I think there are some issues with OS version support:
- With the Windows 11 taskbar, there is no window representing the clock button, so I don't think the code would work.
- On my Windows 10 VM, the clock button class is `TrayClockWClass` rather than `ClockButton`. I wonder where `ClockButton` came from? I see `TrayClockWClass` on Windows 7 as well.